src/backend/allocator/mod.rs

```diff
@@ -49,5 +49,11 @@ pub trait Allocator<B: Buffer> {
     type Error: std::error::Error;
 
     /// Try to create a buffer with the given dimensions and pixel format
-    fn create_buffer(&mut self, width: u32, height: u32, format: Format) -> Result<B, Self::Error>;
+    fn create_buffer(
+        &mut self,
+        width: u32,
+        height: u32,
+        fourcc: Fourcc,
+        modifiers: impl Iterator<Item = Modifier>,
```

Comment:
Why an `impl Iterator` rather than a `&[Modifier]`? It seems that your main implementation starts by collecting in into a `Vec` anyway. Giving a `&[Modifier]` as argument would allow to have lees copying around, no? Or is there something more I'm missing?

Reply:
Just to be more flexible. The `gbm`-api also takes an Iterator, but since we might need to call multiple gbm functions, we cannot really use it.
So I guess, I can change this, good idea.

Reply:
done